_posts/2019-06-07-conventions-and-rubocop.markdown

```diff
@@ -0,0 +1,112 @@
+---
+layout: post
+title: "Code Conventions and Rubocop"
+date: 2019-06-07 09:00:00
+categories: ["ruby", "best-practices"]
+author: "rdormer"
+---
+
+Everyone has had the experience of working on a gnarly, difficult to
+understand code-base.  The sort of code base that makes you hate your
+job.  Often it comes down to poor design, but [code
+conventions](https://en.wikipedia.org/wiki/Coding_conventions) also
+play a large part in wether you wake up dreading your job in the
+morning.  The overall design (choice of design patterns and how
+modules and classes are organized and factored) is the long range, big
+picture strategy of how an application will be made.  Code
+conventions, by contrast, come down to the choices you make about
+which constructs of a language you use, which you don't, and when.
+
+<!--more-->
+
+How you name variables, functions, classes, and how you organize and
+implement control logic determine a lot of the day to day nitty gritty
+of getting your brain dirty and confused.  By themselves, conventions
+won't necessarily derail a project the way that a poorly chosen design
+might, but they can easily end up chewing up time - and therefore
+money.
+
+# Choose Wisely
+
+The main arguments for code conventions come down to readability.  By
+enforcing a standard style, code becomes more uniform.  Code that's
+more uniform becomes easier to read and understand once you get used
+to the conventions - and readability is hugely important, since the
+author of a piece of code will probably have the *least* interaction
+with it over the life of that code.  But more than that, well chosen
+conventions make writing code easier by *removing* the number of
+choices you can make while writing code - even if you don't like the
+idea of being restricted, it's unquestionably easier to write code
+when you don't have to spend a lot of time thinking about how to name
+variables or classes (for instance) because there's really only one
+way you're allowed to do it.
+
+Of course, developers love to disagree, and any conventions like this
+will inevitably be a potential source of griping.  From experience,
+it'll be a fraction of the complaining you'll hear if you *don't* have
+standard conventions, though.  Even the most opinionated developers
+will be professional enough to get used to it quickly enough.  Wether
+they admit it or not, most experienced developers would rather have a
+convention they're not crazy about than no conventions at all.
```

Comment:
@rdormer Maybe you could add a paragraph about discussions in code reviews? I believe having an automated tool for this we will save time in code style discussions.